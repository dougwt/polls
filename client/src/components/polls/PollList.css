.empty {
  font-style: italic;
  color: #999;
  font-size: 0.9em;
  padding: 1.5em !important;
}

.pagination li.active {
  background-color: #06bad6;
}
