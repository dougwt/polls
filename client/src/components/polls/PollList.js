import React from 'react';
import PropTypes from 'prop-types';
import {
  Row,
  Collection,
  CollectionItem,
  Badge,
  Pagination
} from 'react-materialize';

const renderPolls = polls => {
  if (polls.length < 1) {
    return (
      <CollectionItem className="center-align">
        There are no polls to show. Why don&#39;t you create one?
      </CollectionItem>
    );
  }

  return polls.map(poll => {
    return (
      <CollectionItem href={`/polls/${poll.id}`} key={poll.id}>
        {poll.question}
        <Badge>1</Badge>
      </CollectionItem>
    );
  });
};

const PollList = ({ title, polls = [] }) => {
  return (
    <div>
      <Row>
        <h6 className="header section-title">{title}</h6>
      </Row>

      <Row>
        <Collection>{renderPolls(polls)}</Collection>
      </Row>

      <Row className="center-align">
        <Pagination items={10} activePage={2} maxButtons={8} />
      </Row>
    </div>
  );
};
PollList.propTypes = {
  title: PropTypes.string.isRequired,
  polls: PropTypes.array
};

export default PollList;
