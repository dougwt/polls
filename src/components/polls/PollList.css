.empty {
  font-style: italic;
  color: #999;
  font-size: 0.9em;
  padding: 1.5em !important;
}

.pagination li.active {
  background-color: #06bad6;
}

@media screen and (max-width: 600px) {
  .badge,
  .header span {
    display: none;
  }
}
