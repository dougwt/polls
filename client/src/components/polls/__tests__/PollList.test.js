import React from 'react';
import { shallow } from 'enzyme';
import PollList from '../PollList';

describe('PollList', () => {
  let props = { title: 'Test Title' };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PollList {...props} />);
  });

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains a section title', () => {
    expect(wrapper.find('.section-title').exists()).toBe(true);
  });

  it('contains a list of polls', () => {
    expect(wrapper.find('Collection').exists()).toBe(true);
  });

  it('contains a Pagination', () => {
    expect(wrapper.find('Pagination').exists()).toBe(true);
  });

  describe('0 polls', () => {
    beforeEach(() => {
      props = { title: 'Test Title', polls: [] };
      wrapper = shallow(<PollList {...props} />);
    });

    it('shows empty list message', () => {
      expect(wrapper.find('Collection').html()).toContain(
        'There are no polls to show. Why don&#x27;t you create one?'
      );
    });
  });

  describe('1 polls', () => {
    beforeEach(() => {
      props = {
        title: 'Test Title',
        polls: [{ id: 1, question: 'Question 1' }]
      };
      wrapper = shallow(<PollList {...props} />);
    });

    it('shows a single poll', () => {
      expect(wrapper.find('CollectionItem').length).toEqual(1);
    });

    it('shows the title of each poll', () => {
      props.polls.forEach(poll => {
        expect(wrapper.find('CollectionItem').html()).toContain(poll.question);
      });
    });
  });

  describe('2 polls', () => {
    beforeEach(() => {
      props = {
        title: 'Test Title',
        polls: [
          { id: 1, question: 'Question 1' },
          { id: 2, question: 'Question 2' }
        ]
      };
      wrapper = shallow(<PollList {...props} />);
    });

    it('shows two polls', () => {
      expect(wrapper.find('CollectionItem').length).toEqual(2);
    });

    it('shows the title of each poll', () => {
      props.polls.forEach((poll, i) => {
        expect(
          wrapper
            .find('CollectionItem')
            .at(i)
            .html()
        ).toContain(poll.question);
      });
    });

    it('shows a link to view each poll', () => {
      props.polls.forEach((poll, i) => {
        expect(
          wrapper
            .find('CollectionItem')
            .at(i)
            .html()
        ).toContain(`href="/polls/${poll.id}"`);
      });
    });
  });

  describe('10 polls', () => {
    beforeEach(() => {
      props = {
        title: 'Test Title',
        polls: [
          { id: 1, question: 'Question 1' },
          { id: 2, question: 'Question 2' },
          { id: 3, question: 'Question 3' },
          { id: 4, question: 'Question 4' },
          { id: 5, question: 'Question 5' },
          { id: 6, question: 'Question 6' },
          { id: 7, question: 'Question 7' },
          { id: 8, question: 'Question 8' },
          { id: 9, question: 'Question 9' },
          { id: 10, question: 'Question 10' }
        ]
      };
      wrapper = shallow(<PollList {...props} />);
    });

    it('shows ten polls', () => {
      expect(wrapper.find('CollectionItem').length).toEqual(10);
    });

    it('shows the title of each poll', () => {
      props.polls.forEach((poll, i) => {
        expect(
          wrapper
            .find('CollectionItem')
            .at(i)
            .html()
        ).toContain(poll.question);
      });
    });

    it('shows a link to view each poll', () => {
      props.polls.forEach((poll, i) => {
        expect(
          wrapper
            .find('CollectionItem')
            .at(i)
            .html()
        ).toContain(`href="/polls/${poll.id}"`);
      });
    });
  });

  describe('20 polls', () => {
    beforeEach(() => {
      props = {
        title: 'Test Title',
        polls: [
          { id: 1, question: 'Question 1' },
          { id: 2, question: 'Question 2' },
          { id: 3, question: 'Question 3' },
          { id: 4, question: 'Question 4' },
          { id: 5, question: 'Question 5' },
          { id: 6, question: 'Question 6' },
          { id: 7, question: 'Question 7' },
          { id: 8, question: 'Question 8' },
          { id: 9, question: 'Question 9' },
          { id: 10, question: 'Question 10' },
          { id: 11, question: 'Question 11' },
          { id: 12, question: 'Question 12' },
          { id: 13, question: 'Question 13' },
          { id: 14, question: 'Question 14' },
          { id: 15, question: 'Question 15' },
          { id: 16, question: 'Question 16' },
          { id: 17, question: 'Question 17' },
          { id: 18, question: 'Question 18' },
          { id: 19, question: 'Question 19' },
          { id: 20, question: 'Question 20' }
        ]
      };
      wrapper = shallow(<PollList {...props} />);
    });

    it('shows twenty polls', () => {
      expect(wrapper.find('CollectionItem').length).toEqual(20);
    });

    it('shows the title of each poll', () => {
      props.polls.forEach((poll, i) => {
        expect(
          wrapper
            .find('CollectionItem')
            .at(i)
            .html()
        ).toContain(poll.question);
      });
    });

    it('shows a link to view each poll', () => {
      props.polls.forEach((poll, i) => {
        expect(
          wrapper
            .find('CollectionItem')
            .at(i)
            .html()
        ).toContain(`href="/polls/${poll.id}"`);
      });
    });
  });
});
