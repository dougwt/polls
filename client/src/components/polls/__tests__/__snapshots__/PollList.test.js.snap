// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollList renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PollList
    title="Test Title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row>
          <h6
                    className="header section-title"
          >
                    Test Title
                    <span
                              className="right"
                    >
                              Respondents
                    </span>
          </h6>
</Row>,
        <Row>
          <Collection>
                    <li
                              className="center-align collection-item empty"
                    >
                              There are no polls to show. Why don't you create one?
                    </li>
          </Collection>
</Row>,
        <Row
          className="center-align"
>
          <Pagination
                    activePage={1}
                    items={1000}
                    maxButtons={undefined}
                    onSelect={[Function]}
          />
</Row>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h6
            className="header section-title"
>
            Test Title
            <span
                        className="right"
            >
                        Respondents
            </span>
</h6>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Test Title",
              <span
                className="right"
>
                Respondents
</span>,
            ],
            "className": "header section-title",
          },
          "ref": null,
          "rendered": Array [
            "Test Title",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Respondents",
                "className": "right",
              },
              "ref": null,
              "rendered": "Respondents",
              "type": "span",
            },
          ],
          "type": "h6",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Collection>
            <li
                        className="center-align collection-item empty"
            >
                        There are no polls to show. Why don't you create one?
            </li>
</Collection>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <li
              className="center-align collection-item empty"
>
              There are no polls to show. Why don't you create one?
</li>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "There are no polls to show. Why don't you create one?",
              "className": "center-align collection-item empty",
            },
            "ref": null,
            "rendered": "There are no polls to show. Why don't you create one?",
            "type": "li",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Pagination
            activePage={1}
            items={1000}
            maxButtons={undefined}
            onSelect={[Function]}
/>,
          "className": "center-align",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activePage": 1,
            "items": 1000,
            "maxButtons": undefined,
            "onSelect": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row>
            <h6
                        className="header section-title"
            >
                        Test Title
                        <span
                                    className="right"
                        >
                                    Respondents
                        </span>
            </h6>
</Row>,
          <Row>
            <Collection>
                        <li
                                    className="center-align collection-item empty"
                        >
                                    There are no polls to show. Why don't you create one?
                        </li>
            </Collection>
</Row>,
          <Row
            className="center-align"
>
            <Pagination
                        activePage={1}
                        items={1000}
                        maxButtons={undefined}
                        onSelect={[Function]}
            />
</Row>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h6
              className="header section-title"
>
              Test Title
              <span
                            className="right"
              >
                            Respondents
              </span>
</h6>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Test Title",
                <span
                  className="right"
>
                  Respondents
</span>,
              ],
              "className": "header section-title",
            },
            "ref": null,
            "rendered": Array [
              "Test Title",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Respondents",
                  "className": "right",
                },
                "ref": null,
                "rendered": "Respondents",
                "type": "span",
              },
            ],
            "type": "h6",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Collection>
              <li
                            className="center-align collection-item empty"
              >
                            There are no polls to show. Why don't you create one?
              </li>
</Collection>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <li
                className="center-align collection-item empty"
>
                There are no polls to show. Why don't you create one?
</li>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "There are no polls to show. Why don't you create one?",
                "className": "center-align collection-item empty",
              },
              "ref": null,
              "rendered": "There are no polls to show. Why don't you create one?",
              "type": "li",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Pagination
              activePage={1}
              items={1000}
              maxButtons={undefined}
              onSelect={[Function]}
/>,
            "className": "center-align",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activePage": 1,
              "items": 1000,
              "maxButtons": undefined,
              "onSelect": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
